### PR TITLE
Make sure everything has examples

### PR DESCRIPTION
**Pull request by davidpdrsn**
Fixes https://github.com/tower-rs/tower-http/issues/55

### CODE REVIEW

**Review by LucioFranco — APPROVED** [review]

File: tower-http/src/add_extension.rs [hunk 1]
```diff
@@ -1,6 +1,54 @@
...
+//! };
+//!
+//! let mut service = ServiceBuilder::new()
+//!     // Share an `Arc<State>` with all requests.
+//!     .layer(AddExtensionLayer::new(Arc::new(state)))
+//!     .service(service_fn(handle));
+//!
+//! // Call the service.
+//! let response = service
+//!     .ready_and()
```
Comment (davidbarsky) [inline tower-http/src/add_extension.rs:45]:
```suggestion
//!     .ready()
```

File: tower-http/src/compression/mod.rs [hunk 2]
```diff
@@ -1,4 +1,67 @@
...
+//!     .layer(CompressionLayer::new())
+//!     .service(service_fn(handle));
+//!
+//! // Call the service.
+//! let request = Request::builder()
+//!     .header(ACCEPT_ENCODING, "gzip")
+//!     .body(Body::empty())?;
+//!
+//! let response = service
+//!     .ready_and()
```
Comment (davidbarsky) [inline tower-http/src/compression/mod.rs:42]:
```suggestion
//!     .ready()
```

File: tower-http/src/decompression/mod.rs [hunk 3]
```diff
@@ -1,4 +1,57 @@
...
+//!     .layer(DecompressionLayer::new())
+//!     .service(service);
+//!
+//! // Call the service.
+//! //
+//! // `DecompressionLayer` takes care of setting `Accept-Encoding`.
+//! let request = Request::new(Body::empty());
+//!
+//! let response = client
+//!     .ready_and()
```
Comment (davidbarsky) [inline tower-http/src/decompression/mod.rs:36]:
```suggestion
//!     .ready()
```

File: tower-http/src/propagate_header.rs [hunk 4]
```diff
@@ -1,4 +1,42 @@
...
+//!     .layer(PropagateHeaderLayer::new(HeaderName::from_static("x-request-id")))
+//!     .service(service_fn(handle));
+//!
+//! // Call the service.
+//! let request = Request::builder()
+//!     .header("x-request-id", "1337")
+//!     .body(Body::empty())?;
+//!
+//! let response = svc
+//!     .ready_and()
```
Comment (davidbarsky) [inline tower-http/src/propagate_header.rs:30]:
```suggestion
//!     .ready()
```

File: tower-http/src/sensitive_header.rs [hunk 5]
```diff
@@ -1,6 +1,40 @@
...
+//!     // Mark the `Authorization` header as sensitive so it doesn't show in logs
+//!     //
+//!     // `SetSensitiveHeaderLayer` will mark the header as sensitive on both the
+//!     // request and response.
+//!     .layer(SetSensitiveHeaderLayer::new(AUTHORIZATION))
+//!     .service(service_fn(handle));
+//!
+//! // Call the service.
+//! let response = service
+//!     .ready_and()
```
Comment (davidbarsky) [inline tower-http/src/sensitive_header.rs:31]:
```suggestion
//!     .ready()
```

**Review by davidbarsky — APPROVED** [review]
Just a few changes to use the non-deprecated `.ready()`.

**Pull request merged**